### PR TITLE
[xdai] Hack Uniswap SDK + include xDAI

### PR DESCRIPTION
**Pull request by anxolin**
**edit 19/01**: I re-visit this PR. Dima was right in his comment, my solution of defining a package.json dependency using relative path doesn't work on a clean install. So I applied his solution and that fixed the issue. 

For some reason, now it was taking precendence of the original SDK to the hacked one. This is why I had to add something to the tsconfig override. 

With this, it's working. 

**What is not included in this PR**: xdai doesn't work yet, cause this PR is all about hacking their SDK without actually using a different package or rewriting all the files from the original Uniswap. 

Expect very soon a new PR on top of this one to add xDAI support

----

This PR is an attempt to make a less intrusive hack than #92 or #90 to include xDAI in the SDK chainId

I make use of the the work of #90 for redeclaring the conflicting types.
To avoid modifying the files, I declare everything in `src/custom/@uniswap/sdk`, then I include an alias for the original uniswap SDK, `@uniswap/sdk-original`, see `package.json`

I still want to explore another option, but it looks like this is working and allow us to control the pieces we hack from the SDK.
Anyways We should make an issue to uniswap asking them to include xDAI network in their list

### CODE REVIEW

**Comment by Velenir** [comment]
Love how you stretched import resolution to the extreme to allow for this hack!! :clap::clap:

File: package.json [hunk 1]
```diff
@@ -123,6 +123,7 @@
   },
   "license": "GPL-3.0-or-later",
   "dependencies": {
-    "@gnosis.pm/gp-v2-contracts": "^0.0.1-alpha.10"
+    "@gnosis.pm/gp-v2-contracts": "^0.0.1-alpha.10",
+    "@uniswap/sdk-original": "./node_modules/@uniswap/sdk"
```
Comment (Velenir) [inline package.json:127]: This won't work for deps installation from clean slate. You need to set up resolutions for TS + webpack instead, without touching deps, or import from node_modules directly.

Comment (W3stside) [reply package.json:127]: Hi Dima! 👋

File: src/custom/@uniswap/sdk/index.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+// export const STAKE = new Token(
+//   ChainId.XDAI,
+//   '0xb7D311E2Eb55F2f68a9440da38e7989210b9A05e',
+//   18,
+//   'STAKE',
+//   'Stake Token on xDai'
+// )
+
+export const WXDAI = new Token(100, '0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d', 18, 'WXDAI', 'Wrapped XDAI')
+// export const WXDAI = new Token(100, '0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d', 18, 'WXDAI', 'Wrapped XDAI')
```
Comment (W3stside) [inline src/custom/@uniswap/sdk/index.ts:94]: ?

File: package.json (hunk 1 above)
Comment (W3stside) [reply package.json:127]: @anxolin something like:
```json
"resolutions": {
  "@uniswap/sdk-original": "@uniswap/sdk",
  },
```

correct me if I'm wrong obviously

**Review by W3stside — CHANGES_REQUESTED** [review]

File: src/custom/@uniswap/sdk/index.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+//   'WETH',
+//   'Wrapped Ether on xDai'
+// )
+// export const STAKE = new Token(
+//   ChainId.XDAI,
+//   '0xb7D311E2Eb55F2f68a9440da38e7989210b9A05e',
+//   18,
+//   'STAKE',
+//   'Stake Token on xDai'
+// )
```
Comment (alfetopito) [inline src/custom/@uniswap/sdk/index.ts:91]: Is this comment left here intentionally?

File: package.json (hunk 1 above)
Comment (Velenir) [reply package.json:127]: Contrary to what you'd expect, `resolutions` don't rename dependencies.
Try 
```suggestion
    "@uniswap/sdk-original": "npm:@uniswap/sdk"
```

Comment (anxolin) [reply package.json:127]: Hei Dima! sorry I've neglected this PR, we have been very busy. Thanks for dropping by this PR, we miss u man! I hope u are good. Come one day to one zoom call to say hi

I'm re-reviewing this solution. I also saw other alternatives if this one ends up being problematic. I'll also try your solution!

Comment (anxolin) [reply package.json]: Nice trick, `npm:` I struggled a bit but now it's working!

File: src/custom/@uniswap/sdk/index.ts (hunk 2 above)
Comment (anxolin) [reply src/custom/@uniswap/sdk/index.ts:94]: Deleted that. Cleanup will be on next PR when I add the xDAI compatibility

File: src/custom/@uniswap/sdk/index.ts (hunk 3 above)
Comment (anxolin) [reply src/custom/@uniswap/sdk/index.ts:91]: I left a TODO. I'll do a second PR now with just the part of adding xDAI compatibility in the APP. This was more to hack the SDK

**Review by alfetopito — CHANGES_REQUESTED** [review]
Build/tests failed

```
FAIL src/state/transactions/reducer.test.ts

  ● Test suite failed to run

    Cannot find module '@uniswap/sdk' from 'reducer.test.ts'

    > 1 | import { ChainId } from '@uniswap/sdk'

        | ^

      2 | import { createStore, Store } from 'redux'

      3 | import { addTransaction, checkedTransaction, clearAllTransactions, finalizeTransaction } from './actions'

      4 | import reducer, { initialState, TransactionState } from './reducer'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:259:17)

      at Object.<anonymous> (src/state/transactions/reducer.test.ts:1:1)
```

Does this mean that you have to change the import everywhere to point to our custom index now?

**Comment by anxolin** [comment]
> Build/tests failed

YEs, I saw later. It was working, but only for a time! I'm trying other approach

**Pull request closed without merging**